request.go

```diff
@@ -59,9 +59,14 @@ func requestFromPacket(pkt hasPath) *Request {
 
 // NewRequest creates a new Request object.
 func NewRequest(method, path string) *Request {
-	ctx, cancel := context.WithCancel(context.Background())
-	return &Request{Method: method, Filepath: cleanPath(path),
-		ctx: ctx, cancelCtx: cancel}
+	return &Request{Method: method, Filepath: cleanPath(path)}
+}
+
+// Used internally to create request with default context
+func newRequestWithContext(method, path string) *Request {
```

Comment:
A naming point here. I would expect a `newRequestWithContext` function to have a context.Context as a parameter, which gets folded into the `&Request{}`. In this case, I think `newContextRequest` would make more sense.

But then, the English-usage here is ambiguous anyways, so…

Reply:
I can see how the WithContext part of the newRequestWithContext does seem to imply passing in a context as that is what WithContext does... So sure, this seems like a reasonable change.